src/main/java/fr/socrates/domain/candidate/CandidateServiceImpl.java

```diff
@@ -17,10 +17,21 @@ public CandidateServiceImpl(CandidateRepository candidateRepository) {
     }
 
     @Override
-    public boolean add(Candidate candidate) {
+    public boolean add(Candidate candidate) throws CandidatePersisteDataException, CandidateExistingException {
+        checkThatCandidateDoesntExist(candidate);
+        checkSuccessfulCandidateSave(candidate);
+        return true;
```

Comment:
return value is always true here. It's not used, you could delete it as you handle errors with exceptions